src/riscv/lib/src/jit/state_access.rs

```diff
@@ -805,6 +880,72 @@ impl<'a, MC: MemoryConfig, M: JitStateAccess> JsaCalls<'a, MC, M> {
         let call = builder.ins().call(*reservation_set_read, &[core_ptr]);
         X64(builder.inst_results(call)[0])
     }
+
+    /// Emit the required IR to call `f64_from_x64_unsigned_dynamic`.
+    ///
+    /// Returns `errno` - on success, the new F64 value is returned.
+    pub(super) fn f64_from_x64_unsigned_dynamic(
+        &mut self,
+        builder: &mut FunctionBuilder<'_>,
+        core_ptr: Value,
+        xval: X64,
+    ) -> impl Errno<F64, MC, M> + 'static {
+        let exception_slot = self.exception_ptr_slot(builder);
+        let exception_ptr = exception_slot.ptr(builder);
+
+        let f64_slot = self.f64_ptr_slot(builder);
+        let f64_ptr = f64_slot.ptr(builder);
+
+        let new_f64_from_x64_unsigned_dynamic =
+            self.f64_from_x64_unsigned_dynamic.get_or_insert_with(|| {
+                self.module
+                    .declare_func_in_func(self.imports.f64_from_x64_unsigned_dynamic, builder.func)
+            });
+
+        let call = builder.ins().call(*new_f64_from_x64_unsigned_dynamic, &[
+            core_ptr,
+            exception_ptr,
+            xval.0,
+            f64_ptr,
+        ]);
+
+        let errno = builder.inst_results(call)[0];
+        ErrnoImpl::new(errno, exception_ptr, move |builder| {
+            // Safety: the fval_slot is initialised in the success case before this closure is called,
```

Comment:
Lets follow Rust guidelines on capitalisation for the comments.

https://std-dev-guide.rust-lang.org/policy/safety-comments.html#inside-safe-elements

```suggestion
            // SAFETY: the fval_slot is initialised in the success case before this closure is called,
```